packages/openapi3/test/emit-openapi.test.ts

```diff
@@ -1,79 +1,136 @@
+    info:
+      title: (title)
+      version: 0.0.0
+    tags: []
+    paths: {}
+    components:
+      schemas:
+        TestEnum:
+          type: string
+          enum:
+            - 'y'
+            - 'Y'
+            - 'yes'
+            - 'Yes'
+            - 'YES'
+            - yEs
+            - 'n'
+            - 'N'
+            - 'no'
+            - 'No'
+            - 'NO'
+            - nO
+            - 'true'
+            - 'True'
+            - 'TRUE'
+            - tRUE
+            - 'false'
+            - 'False'
+            - 'FALSE'
+            - fALSE
+            - 'on'
+            - 'On'
+            - 'ON'
+            - oN
+            - 'off'
+            - 'Off'
+            - 'OFF'
+            - oFF
+    `);
+    });
+
+    interface ServiceNameCase {
+      description: string;
+      code: string;
+      outputFilePattern: string;
+      expectedOutputFiles: string[];
+    }
+    it.each([
+      // {service-name} cases
+      {
+        description: "{service-name} for one service",
+        code: "@service namespace AAA { model M {a: string} }",
+        outputFilePattern: "{service-name}.yaml",
+        expectedOutputFiles: [resolveVirtualPath("AAA.yaml")],
+      },
+      {
+        description: "{service-name} for multiple services",
+        code:
+          "@service namespace AAA { model M {a: string} }" +
+          "@service namespace BBB { model N {b: string} }",
+        outputFilePattern: "{service-name}.yaml",
+        expectedOutputFiles: [resolveVirtualPath("AAA.yaml"), resolveVirtualPath("BBB.yaml")],
+      },
+      // {service-name-if-multiple} cases
+      {
+        description: "{service-name-if-multiple} for one service",
+        code: "@service namespace AAA { model M {a: string} }",
+        outputFilePattern: "{service-name-if-multiple}.yaml",
+        expectedOutputFiles: [resolveVirtualPath("yaml")],
+      },
+      {
+        description: "{service-name-if-multiple} for multiple services",
+        code:
+          "@service namespace AAA { model M {a: string} }" +
+          "@service namespace BBB { model N {b: string} }",
+        outputFilePattern: "{service-name-if-multiple}.yaml",
+        expectedOutputFiles: [resolveVirtualPath("AAA.yaml"), resolveVirtualPath("BBB.yaml")],
+      },
+      // fixed name cases
+      {
+        description: "fixed name for one service",
+        code: "@service namespace AAA { model M {a: string} }",
+        outputFilePattern: "fixed-name.yaml",
+        expectedOutputFiles: [resolveVirtualPath("fixed-name.yaml")],
+      },
+    ])("$description", async (c: ServiceNameCase) => {
+      const options = {
+        "output-file": c.outputFilePattern,
+        "emitter-output-dir": "{output-dir}",
+      };
+      const [diag, load, _] = await helper.emitOpenApi(c.code, options);
+      expect(diag.length).toBe(0);
+      for (const outputFile of c.expectedOutputFiles) expect(load(outputFile)).toBeDefined();
```

Comment:
instead of this new helper just check the `fs` directly here

Reply:
none of the helper returns `fs`